source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1461 @@
+
+  - ``readPreference``: Optional document. See `commonOptions_readPreference`_.
+
+  - ``writeConcern``: Optional document. See `commonOptions_writeConcern`_.
+
+
+Common Options
+~~~~~~~~~~~~~~
+
+This section defines the structure of common options that are referenced from
+various contexts in the test format. Comprehensive documentation for some of
+these types and their parameters may be found in the following specifications:
+
+- `Read and Write Concern <../read-write-concern/read-write-concern.rst>`__.
+- `Server Selection: Read Preference <../server-selection/server-selection.rst#read-preference>`__.
+
+The structure of these common options is as follows:
+
+.. _commonOptions_readConcern:
+
+- ``readConcern``: Optional document. Map of parameters to construct a read
+  concern. The structure of this document is as follows:
+
+  - ``level``: Required string.
+
+.. _commonOptions_readPreference:
+
+- ``readPreference``: Optional document. Map of parameters to construct a read
+  preference. The structure of this document is as follows:
+
+  - ``mode``: Required string.
+
+  - ``tagSets``: Optional array of documents.
+
+  - ``maxStalenessSeconds``: Optional integer.
+
+  - ``hedge``: Optional document.
+
+.. _commonOptions_session:
+
+- ``session``: Optional string. Session entity which will be resolved to a
+  ClientSession object. The YAML file SHOULD use an `alias node`_ for a session
+  entity's ``id`` field (e.g. ``session: *session0``).
+
+.. _commonOptions_writeConcern:
+
+- ``writeConcern``: Optional document. Map of parameters to construct a write
+  concern. The structure of this document is as follows:
+
+  - ``journal``: Optional boolean.
+
+  - ``w``: Optional integer or string.
+
+  - ``wtimeoutMS``: Optional integer.
+
+
+Version String
+--------------
+
+Version strings, which are used for `schemaVersion`_ and `runOn`_, MUST conform
+to one of the following formats, where each component is an integer:
+
+- ``<major>.<minor>.<patch>``
+- ``<major>.<minor>`` (``<patch>>`` is assumed to be zero)
+- ``<major>`` (``<minor>`` and ``<patch>>`` are assumed to be zero)
+
+
+Entity Test Operations
+----------------------
+
+Most operations correspond to an API method on a driver object. If
+`operation.object <operation_object_>`_ refers to an `entity`_ name (e.g.
+"collection0") then `operation.name <operation_name_>`_ is expected to reference
+an API method on that class. Required and optional parameters for API methods
+are both specified directly within `operation.arguments <operation_arguments_>`_
+(e.g. ``upsert`` for ``updateOne`` is *not* nested under an ``options`` key).
+
+This spec does not provide exhaustive documentation for all possible API methods
+that may appear in a test; however, the following sections discuss all supported
+entities and their operations in some level of detail.
+
+**TODO**: While CRUD methods tend to flatten options into ``arguments``, session
```

Comment:
I added language to clarify that `options` keys must not be used. AFAIK, no specs have naming conflicts between required and optional parameters. This will also remove a long-standing inconsistency between some of the CRUD tests (where regular methods didn't agree with how we structure write models for `bulkWrites`).